libs/application-generic/src/usecases/get-subscriber-global-preference/get-subscriber-global-preference.usecase.ts

```diff
@@ -6,78 +6,121 @@ import {
+import { filteredPreference } from '../get-subscriber-template-preference/get-subscriber-template-preference.usecase';
 
 @Injectable()
 export class GetSubscriberGlobalPreference {
   constructor(
     private subscriberPreferenceRepository: SubscriberPreferenceRepository,
     private subscriberRepository: SubscriberRepository,
     private getPreferences: GetPreferences,
+    private getSubscriberPreference: GetSubscriberPreference,
   ) {}
 
   async execute(command: GetSubscriberGlobalPreferenceCommand) {
-    const subscriber =
-      command.subscriber ??
-      (await this.fetchSubscriber({
-        subscriberId: command.subscriberId,
-        _environmentId: command.environmentId,
-      }));
+    const subscriber = await this.getSubscriber(command);
 
-    if (!subscriber) {
-      throw new ApiException(`Subscriber ${command.subscriberId} not found`);
-    }
+    const activeChannels = await this.getActiveChannels(command);
 
-    const subscriberPreference =
-      await this.subscriberPreferenceRepository.findOne({
-        _environmentId: command.environmentId,
-        _subscriberId: subscriber._id,
-        level: PreferenceLevelEnum.GLOBAL,
-      });
+    const subscriberGlobalPreference = await this.getSubscriberGlobalPreference(
+      command,
+      subscriber._id,
+    );
 
-    const subscriberChannelPreference =
-      (await this.getPreferences.getPreferenceChannels({
-        environmentId: command.environmentId,
-        organizationId: command.organizationId,
-        subscriberId: subscriber._id,
-      })) || subscriberPreference?.channels;
-    const channels = this.updatePreferenceStateWithDefault(
-      subscriberChannelPreference ?? {},
+    const channelsWithDefaults = this.buildDefaultPreferences(
+      subscriberGlobalPreference,
     );
 
+    const channels = filteredPreference(channelsWithDefaults, activeChannels);
+
     return {
       preference: {
-        enabled: subscriberPreference?.enabled ?? true,
+        enabled: true,
         channels,
       },
     };
   }
 
+  private async getSubscriberGlobalPreference(
+    command: GetSubscriberGlobalPreferenceCommand,
+    subscriberId: string,
+  ): Promise<IPreferenceChannels> {
+    /** @deprecated */
+    const subscriberGlobalPreferenceV1 =
+      await this.subscriberPreferenceRepository.findOne({
+        _environmentId: command.environmentId,
+        _subscriberId: subscriberId,
+        level: PreferenceLevelEnum.GLOBAL,
+      });
+
+    const subscriberGlobalPreferenceV2 =
+      await this.getPreferences.getPreferenceChannels({
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        subscriberId,
+      });
+
+    const subscriberGlobalPreference =
+      subscriberGlobalPreferenceV2 ??
+      subscriberGlobalPreferenceV1?.channels ??
+      {};
+
+    return subscriberGlobalPreference;
+  }
+
+  private async getActiveChannels(
```

Comment:
This is the change in Subscriber Global Preference fetching that is responsible for ensuring that only active channels are shown in the Inbox/NotificationCenter preferences.